python-avd/pyavd/_eos_designs/structured_config/underlay/router_pim_sparse_mode.py

```diff
@@ -19,53 +20,38 @@ class RouterPimSparseModeMixin(Protocol):
     Class should only be used as Mixin to a AvdStructuredConfig class.
     """
 
-    @cached_property
-    def router_pim_sparse_mode(self: AvdStructuredConfigUnderlayProtocol) -> dict | None:
+    @structured_config_contributor
+    def router_pim_sparse_mode(self: AvdStructuredConfigUnderlayProtocol) -> None:
         """
-        Return structured config for router_pim_sparse_mode.
+        Set the structured config for router_pim_sparse_mode.
 
         Used for to configure multicast RPs for the underlay
         """
         if not self.shared_utils.underlay_multicast or not self.inputs.underlay_multicast_rps:
-            return None
+            return
 
-        rp_addresses = []
-        anycast_rps = []
         for rp_entry in self.inputs.underlay_multicast_rps:
-            rp_address = {"address": rp_entry.rp}
+            rp_address = EosCliConfigGen.RouterPimSparseMode.Ipv4.RpAddressesItem(address=rp_entry.rp)
             if rp_entry.groups:
                 if rp_entry.access_list_name:
-                    rp_address["access_lists"] = [rp_entry.access_list_name]
+                    rp_address.access_lists.append(rp_entry.access_list_name)
                 else:
-                    rp_address["groups"] = rp_entry.groups._as_list()
+                    for group in rp_entry.groups:
+                        rp_address.groups.append(group)
```

Comment:
Given that we are manipulating two list of strings I think this should work:
```suggestion
                    rp_address.groups.extend(rp._entry.groups)
```